sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,298 @@
+			break;
+		}
+		timestamp = tmp;
+		actField = nextField;
+
+		if (timestamp < rd->timeLast && rd->timeLast != READER_TIMESTAMP_NOT_SET) {
+			rd->timeOffset += rd->timeLast - timestamp;
+		}
+		rd->timeLast = timestamp;
+
+		timestamp += rd->timeOffset;
+
+		if (timeStart == READER_TIMESTAMP_NOT_SET) {
+			timeStart = timestamp;
+		}
+
+		if (timestamp - timeStart > rd->timeHorizon) {
+			/* Go to start of the line */
+			if (fseek(rd->scenarioFile, -lineLen, SEEK_CUR) != 0) {
+				err = -1;
+			}
+			break;
+		}
+
+		switch (sensorID) {
+			case SENSOR_TYPE_ACCEL:
+				err = reader_accelDataParse(actField, timestamp, &parsed);
+				break;
+			case SENSOR_TYPE_BARO:
+				err = reader_baroDataParse(actField, timestamp, &parsed);
+				break;
+			default:
+				fprintf(stderr, "%s: Unknown sensor type\n", __FUNCTION__);
+				err = -1;
+				break;
+		}
+
+		if (err < 0) {
+			break;
+		}
+
+		if (eventQueue_enqueue(queue, &parsed) < 0) {
+			err = -1;
+			break;
+		}
+	}
+
+	return (err < 0) ? err : 0;
+}
+
+
+void reader_close(simsens_reader_t *reader)
+{
+	if (reader == NULL) {
+		return;
+	}
+
+	if (reader->scenarioFile != NULL) {
+		fclose(reader->scenarioFile);
+	}
+
+	free(reader->lineBuf);
+}
+
+
+int reader_open(simsens_reader_t *rd, const char *path, int sensorTypes, time_t timeHorizon)
+{
+	ssize_t lineLen, readBytes;
+
+	if (rd == NULL || path == NULL) {
+		return -1;
+	}
+
+	rd->scenarioFile = fopen(path, "r");
+	if (rd->scenarioFile == NULL) {
+		return -1;
+	}
+
+	rd->lineBuf = NULL;
+	rd->bufLen = 0;
+	rd->sensorTypes = sensorTypes;
+	rd->timeHorizon = timeHorizon;
+	rd->timeLast = READER_TIMESTAMP_NOT_SET;
+
+	lineLen = getline(&rd->lineBuf, &rd->bufLen, rd->scenarioFile);
+	if (lineLen < 1) {
+		free(rd->lineBuf);
+		fclose(rd->scenarioFile);
+		return -1;
+	}
+
+	/* Checking if current line is header */
+	if (rd->lineBuf[0] == 's' || rd->lineBuf[0] == 'S') {
+		rd->headerLen = lineLen;
+	}
+	else {
+		rd->headerLen = 0;
+
+		/* If there is no header go to begging of the file */
+		if (fseek(rd->scenarioFile, -lineLen, SEEK_CUR) != 0) {
+			free(rd->lineBuf);
+			fclose(rd->scenarioFile);
+			return -1;
+		}
+
+		clearerr(rd->scenarioFile);
+	}
+
+	/* Omitting first sensorID */
+	lineLen = getdelim(&rd->lineBuf, &rd->bufLen, READER_DATA_SEPARATOR, rd->scenarioFile);
+	if (lineLen < 1) {
+		free(rd->lineBuf);
+		fclose(rd->scenarioFile);
+		return -1;
+	}
+	readBytes = lineLen;
+
+	/* Setting up timeOffset init value */
+	lineLen = getdelim(&rd->lineBuf, &rd->bufLen, READER_DATA_SEPARATOR, rd->scenarioFile);
+	if (lineLen < 1) {
+		free(rd->lineBuf);
+		fclose(rd->scenarioFile);
+		return -1;
+	}
+	readBytes += lineLen;
+
+	if (reader_getFieldLLong(rd->lineBuf, NULL, &rd->timeOffset) != 0) {
+		free(rd->lineBuf);
+		fclose(rd->scenarioFile);
+		return -1;
+	}
+	rd->timeOffset = -rd->timeOffset;
+
+	if (fseek(rd->scenarioFile, -readBytes, SEEK_CUR) != 0) {
+		free(rd->lineBuf);
+		fclose(rd->scenarioFile);
+		return -1;
+	}
```

Comment:
My suggestion, a bit long.
```suggestion
	rd->headerLen = 0;
	/* Checking if current line is header */
	if (rd->lineBuf[0] == 's' || rd->lineBuf[0] == 'S') {
		/* Read next line */
		if (getline(&rd->lineBuf, &rd->bufLen, rd->scenarioFile) < 1) {
			free(rd->lineBuf);
			fclose(rd->scenarioFile);
			return -1;
		}
		rd->headerLen = lineLen;
	}

	/* Omitting first sensorID */
	char *p = strchr(rd->lineBuf, READER_DATA_SEPARATOR);
	if ((p == NULL) ||  (reader_getFieldLLong(rd->lineBuf, NULL, &rd->timeOffset) != 0) {
		free(rd->lineBuf);
		fclose(rd->scenarioFile);
		return -1;
	}
	rd->timeOffset = -rd->timeOffset;

	if (fseek(rd->scenarioFile, rd->headerLen, SEEK_SET) != 0) {
		free(rd->lineBuf);
		fclose(rd->scenarioFile);
		return -1;
	}
```
In addition:
```c
#include <string.h>
```
remove variable `read_bytes`